dandi/dandiapi.py

```diff
@@ -436,19 +436,21 @@ def publish(self) -> "RemoteDandiset":
         `Version`.
         """
         lgr.debug("Waiting for Dandiset %s to complete validation ...", self.identifier)
-        while True:
+        start = time()
+        # Wait 20s for celery tasks to finish
+        while time() - start < 20:
             r = self.client.get(f"{self.version_api_path}info/")
             if "status" not in r:
                 # Running against older version of dandi-api that doesn't
                 # validate
                 break
             if r["status"] == "Valid":
                 break
-            elif r["status"] == "Invalid":
-                raise ValueError(
-                    f"Dandiset {self.identifier} is invalid: {r['validation_error']}"
-                )
             sleep(0.5)
+        else:
+            raise ValueError(
+                f"Dandiset {self.identifier} is {r['status']}: {r['validation_error']}"
+            )
```

Comment:
should it ideally result in e.g. 503 with `Retry-After` set for e.g. a few minutes if it is a matter of celery job just not completing within 20 seconds and otherwise not clear status ?

Reply:
The server doesn't necessarily know that another validation is in the queue, so it might not be appropriate to always respond with a 503. RabbitMQ does not allow non-destructive access to the message queue, so there's no real way to tell that another validation is coming soon.